pkg/sql/catalog/tabledesc/constraint.go

```diff
@@ -11,11 +11,204 @@
 package tabledesc
 
 import (
+	"fmt"
+
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/errors"
 )
 
+type constraintBase struct {
+	maybeMutation
+}
+
+// AsCheck implements the catalog.Constraint interface.
+func (c constraintBase) AsCheck() catalog.CheckConstraint {
+	return nil
+}
+
+// AsForeignKey implements the catalog.Constraint interface.
+func (c constraintBase) AsForeignKey() catalog.ForeignKeyConstraint {
+	return nil
+}
+
+// AsUniqueWithoutIndex implements the catalog.Constraint interface.
+func (c constraintBase) AsUniqueWithoutIndex() catalog.UniqueWithoutIndexConstraint {
+	return nil
+}
+
+// AsUnique implements the catalog.Constraint interface.
+func (c constraintBase) AsUnique() catalog.UniqueWithIndexConstraint {
+	return nil
+}
+
+type checkConstraint struct {
+	constraintBase
+	desc            *descpb.TableDescriptor_CheckConstraint
+	notNullColumnID descpb.ColumnID
+}
+
+var _ catalog.CheckConstraint = (*checkConstraint)(nil)
+
+// CheckDesc implements the catalog.CheckConstraint interface.
+func (c checkConstraint) CheckDesc() *descpb.TableDescriptor_CheckConstraint {
+	return c.desc
+}
+
+// Expr implements the catalog.CheckConstraint interface.
+func (c checkConstraint) Expr() string {
+	return c.desc.Expr
+}
+
+// NumReferencedColumns implements the catalog.CheckConstraint interface.
+func (c checkConstraint) NumReferencedColumns() int {
+	if c.notNullColumnID != 0 {
+		return 1
+	}
+	return len(c.desc.ColumnIDs)
```

Comment:
I guess I asked the question above: if `c.notNullColumnID != 0`, do we assure `len(c.desc.ColumnIDs)` is also 1? If not, that will be non-sensical.

Reply:
I've since rewritten this.